### ISSUE TITLE
[4.0] Check Session token in header

### ISSUE THREAD

**Issue by schnuti** [joomla/joomla-cms#20642]
### Steps to reproduce the issue
Was there a reason to remove this check from checkToken() in Session.php?

```		// Check from header first
		if ($token === $app->input->server->get('HTTP_X_CSRF_TOKEN', '', 'alnum'))
		{
			return true;
		}
```

It's still handled and set in core.js - function Joomla.request

One of them is a bug.

**Comment by ghost** [comment]
Can an more experienced User than me please answer as this Issue is set on "New" since 11 Days?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20642">issues.joomla.org/tracker/joomla-cms/20642</a>.</sub>

**Comment by mbabker** [comment]
Probably lost in a merge conflict, not intentionally removed.

TBH, it's time to deprecate the static methods in the CMS Session class and we should look at how to transition to the [application's `checkToken` method](https://github.com/joomla-framework/application/blob/ca4c5a10a0efe7c2d9d413a05572ee26b1f7b6f4/src/AbstractWebApplication.php#L1012-L1038).

**Comment by ghost** [comment]
thanks for Comment, @mbabker

**Comment by brianteeman** [comment]
closed see #20786
